lib/suppose-stream.js

```diff
@@ -55,7 +56,7 @@ function SupposeStream(options) {
     switch(typeof condition)
     {
       case 'string':
-        match = buffer.endsWith(condition)
+        match = new RegExp('^[^\S\r\n]*' + escape(condition) + '[^\S\r\n]*$', 'gm').test(buffer);
```

Comment:
Text matcher match against end of the buffer, no worrying about what was before. The check for the whitespaces at beginning of the buffer will force that, so remove it.